Frequent pytest failures in CI build

**Describe the bug**
CI builds have started failing more and more frequently with a pytest failure:
```
##[error]Failed to test Python inside ../src/Python/qsharp-core/qsharp/tests
```

The failure occurs due to an `abort` response status from the IQ# kernel while processing the `%azure.connect` magic command, e.g.:

`DEBUG    qsharp.clients.iqsharp:iqsharp.py:213 sending:`
`%azure.connect {"storage": "test", "subscription": "test", "resourceGroup": "test", "workspace": "test"}`

`DEBUG    qsharp.clients.iqsharp:iqsharp.py:256 received:`
`{'header': {'msg_id': 'c15aee67-cbce-4700-a9b1-6a3acf8a87c3', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_reply', 'version': '5.2.0', 'date': datetime.datetime(2020, 9, 4, 23, 20, 7, 436112, tzinfo=datetime.timezone.utc)}, 'msg_id': 'c15aee67-cbce-4700-a9b1-6a3acf8a87c3', 'msg_type': 'execute_reply', 'parent_header': {'msg_id': '13b75548-ac95ab017606a8fdfecdc565_2', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_request', 'version': '5.3'}, 'metadata': None, 'content': {'status': 'abort', 'execution_count': None, 'payload': [], 'user_expressions': None}, 'buffers': []}`

I have not yet been successful in attempting to reproduce this locally.

**Sample log**
For example, see this log from a recent CI build:
https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=14187&view=logs&j=9e36a536-c213-5644-c2ab-401b20f4b829&t=57baf6f6-621f-5161-f318-42bbd94a0177&l=279

```
2020-09-04T23:21:06.5390751Z =================================== FAILURES ===================================
2020-09-04T23:21:06.5391716Z _____________________________ test_empty_workspace _____________________________
2020-09-04T23:21:06.5392615Z 
2020-09-04T23:21:06.5393041Z     def test_empty_workspace():
2020-09-04T23:21:06.5393499Z         """
2020-09-04T23:21:06.5393988Z         Tests behavior of a mock workspace with no providers.
2020-09-04T23:21:06.5394448Z         """
2020-09-04T23:21:06.5394957Z         with pytest.raises(AzureError) as exception_info:
2020-09-04T23:21:06.5395487Z             qsharp.azure.target()
2020-09-04T23:21:06.5395982Z         assert exception_info.value.error_name == "NotConnected"
2020-09-04T23:21:06.5397134Z     
2020-09-04T23:21:06.5397804Z         targets = qsharp.azure.connect(
2020-09-04T23:21:06.5398435Z             storage="test",
2020-09-04T23:21:06.5398882Z             subscription="test",
2020-09-04T23:21:06.5399312Z             resourceGroup="test",
2020-09-04T23:21:06.5399727Z >           workspace="test"
2020-09-04T23:21:06.5400170Z         )
2020-09-04T23:21:06.5400824Z 
2020-09-04T23:21:06.5401237Z test_azure.py:43: 
2020-09-04T23:21:06.5401776Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2020-09-04T23:21:06.5402216Z 
2020-09-04T23:21:06.5403767Z params = {'resourceGroup': 'test', 'storage': 'test', 'subscription': 'test', 'workspace': 'test'}
2020-09-04T23:21:06.5404814Z result = None
2020-09-04T23:21:06.5405385Z 
2020-09-04T23:21:06.5406444Z     def connect(**params) -> List[AzureTarget]:
2020-09-04T23:21:06.5406867Z         """
2020-09-04T23:21:06.5407443Z         Connects to an Azure Quantum workspace or displays current connection status.
2020-09-04T23:21:06.5408428Z         See https://docs.microsoft.com/qsharp/api/iqsharp-magic/azure.connect for more details.
2020-09-04T23:21:06.5409338Z         """
2020-09-04T23:21:06.5409796Z         result = qsharp.client._execute_magic(f"azure.connect", raise_on_stderr=False, **params)
2020-09-04T23:21:06.5410364Z >       if "error_code" in result: raise AzureError(result)
2020-09-04T23:21:06.5411100Z E       TypeError: argument of type 'NoneType' is not iterable
2020-09-04T23:21:06.5411678Z 
2020-09-04T23:21:06.5411983Z ../azure.py:110: TypeError
2020-09-04T23:21:06.5412695Z ------------------------------ Captured log setup ------------------------------
2020-09-04T23:21:06.5413313Z INFO     qsharp.clients.iqsharp:iqsharp.py:82 Starting IQ# kernel...
2020-09-04T23:21:06.5414482Z DEBUG    traitlets:manager.py:307 Starting kernel: ['/home/vsts/work/1/s/src/Tool/bin/Release/netcoreapp3.1/Microsoft.Quantum.IQSharp', 'kernel', '--log-level', 'Error', '/tmp/tmpgnb3qp1o.json', '--user-agent', 'qsharp.py']
2020-09-04T23:21:06.5415514Z DEBUG    traitlets:connect.py:544 Connecting to: tcp://127.0.0.1:45257
2020-09-04T23:21:06.5415990Z DEBUG    qsharp.clients.iqsharp:iqsharp.py:213 sending:
2020-09-04T23:21:06.5416393Z %version
2020-09-04T23:21:06.5416791Z DEBUG    traitlets:client.py:162 connecting iopub channel to tcp://127.0.0.1:57673
2020-09-04T23:21:06.5417789Z DEBUG    traitlets:connect.py:544 Connecting to: tcp://127.0.0.1:57673
2020-09-04T23:21:06.5418564Z DEBUG    traitlets:client.py:174 connecting stdin channel to tcp://127.0.0.1:60775
2020-09-04T23:21:06.5419192Z DEBUG    traitlets:connect.py:544 Connecting to: tcp://127.0.0.1:60775
2020-09-04T23:21:06.5419820Z DEBUG    traitlets:client.py:150 connecting shell channel to tcp://127.0.0.1:57225
2020-09-04T23:21:06.5420434Z DEBUG    traitlets:connect.py:544 Connecting to: tcp://127.0.0.1:57225
2020-09-04T23:21:06.5420981Z DEBUG    qsharp.clients.iqsharp:iqsharp.py:256 received:
2020-09-04T23:21:06.5424217Z {'header': {'msg_id': '01e92043-c0c3-415f-bfa3-818a099d6d11', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_reply', 'version': '5.2.0', 'date': datetime.datetime(2020, 9, 4, 23, 20, 7, 405487, tzinfo=datetime.timezone.utc)}, 'msg_id': '01e92043-c0c3-415f-bfa3-818a099d6d11', 'msg_type': 'execute_reply', 'parent_header': {'msg_id': '13b75548-ac95ab017606a8fdfecdc565_0', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_request', 'version': '5.3'}, 'metadata': None, 'content': {'status': 'ok', 'execution_count': 1, 'payload': [], 'user_expressions': None}, 'buffers': []}
2020-09-04T23:21:06.5426414Z INFO     qsharp.clients.iqsharp:iqsharp.py:98 Q# version
2020-09-04T23:21:06.5427308Z {'iqsharp': LooseVersion ('0.12.2009.418'), 'Jupyter Core': LooseVersion ('1.4.0.0'), '.NET Runtime': LooseVersion ('.NETCoreApp,Version=v3.1')}
2020-09-04T23:21:06.5428425Z ----------------------------- Captured stderr call -----------------------------
2020-09-04T23:21:06.5429019Z Please call %azure.connect before getting the execution target.
2020-09-04T23:21:06.5429960Z ------------------------------ Captured log call -------------------------------
2020-09-04T23:21:06.5430725Z DEBUG    qsharp.clients.iqsharp:iqsharp.py:213 sending:
2020-09-04T23:21:06.5431162Z %azure.target  {}
2020-09-04T23:21:06.5431620Z DEBUG    qsharp.clients.iqsharp:iqsharp.py:256 received:
2020-09-04T23:21:06.5433999Z {'header': {'msg_id': '54e9ee46-fcac-4ffc-98df-6ecdbac78ed5', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_reply', 'version': '5.2.0', 'date': datetime.datetime(2020, 9, 4, 23, 20, 7, 432986, tzinfo=datetime.timezone.utc)}, 'msg_id': '54e9ee46-fcac-4ffc-98df-6ecdbac78ed5', 'msg_type': 'execute_reply', 'parent_header': {'msg_id': '13b75548-ac95ab017606a8fdfecdc565_1', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_request', 'version': '5.3'}, 'metadata': None, 'content': {'status': 'error', 'execution_count': 2, 'payload': [], 'user_expressions': None}, 'buffers': []}
2020-09-04T23:21:06.5436173Z DEBUG    qsharp.clients.iqsharp:iqsharp.py:213 sending:
2020-09-04T23:21:06.5436731Z %azure.connect {"storage": "test", "subscription": "test", "resourceGroup": "test", "workspace": "test"}
2020-09-04T23:21:06.5437361Z DEBUG    qsharp.clients.iqsharp:iqsharp.py:256 received:
2020-09-04T23:21:06.5439697Z {'header': {'msg_id': 'c15aee67-cbce-4700-a9b1-6a3acf8a87c3', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_reply', 'version': '5.2.0', 'date': datetime.datetime(2020, 9, 4, 23, 20, 7, 436112, tzinfo=datetime.timezone.utc)}, 'msg_id': 'c15aee67-cbce-4700-a9b1-6a3acf8a87c3', 'msg_type': 'execute_reply', 'parent_header': {'msg_id': '13b75548-ac95ab017606a8fdfecdc565_2', 'username': 'vsts', 'session': '13b75548-ac95ab017606a8fdfecdc565', 'msg_type': 'execute_request', 'version': '5.3'}, 'metadata': None, 'content': {'status': 'abort', 'execution_count': None, 'payload': [], 'user_expressions': None}, 'buffers': []}
2020-09-04T23:21:06.5441694Z =========================== short test summary info ============================
2020-09-04T23:21:06.5442480Z FAILED test_azure.py::test_empty_workspace - TypeError: argument of type 'Non...
2020-09-04T23:21:06.5443268Z =================== 1 failed, 17 passed in 77.44s (0:01:17) ====================
2020-09-04T23:21:09.9500946Z ##[error]Failed to test Python inside ../src/Python/qsharp-core/qsharp/tests
```

--- a/src/Jupyter/Jupyter.csproj
+++ b/src/Jupyter/Jupyter.csproj
@@ -35,7 +35,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Jupyter.Core" Version="1.4.80531" />
+    <PackageReference Include="Microsoft.Jupyter.Core" Version="1.4.112145" />
     <PackageReference Include="System.Reactive" Version="4.3.2" />
   </ItemGroup>
 

